onnxruntime/core/providers/cpu/nn/pool_base.h

```diff
@@ -192,16 +192,14 @@ class PoolBase {
           *out_size = (in_size - kernel) / stride + 1;
           break;
         case AutoPadType::SAME_LOWER: {
-          int64_t legacy_target_size = (in_size + stride - 1) / stride;
-          int64_t pad_needed = (legacy_target_size - 1) * stride + kernel - in_size;
+          int64_t pad_needed = (in_size - 1) * stride + kernel - in_size;
```

Comment:
unable to check-in unit tests for pool ops as the implementation does not seem to handle cases wherein an output value is to be generated by a "pure padding" region - it needs deeper investigation and possibly multiple bug fixes to solve that, so tracking the issue in #801. The scope of this PR is to compute the right padding values in case where AutoPad attribute is used (assuming the implementation can handle the computed pad values)